src/main/java/org/wso2/okta/client/OktaOAuthClient.java

```diff
@@ -0,0 +1,1046 @@
+     * @param paramMap             It has additional parameters to create the Json payload.
+     * @return JSON payload.
+     * @throws APIManagementException This is the custom exception class for API management.
+     */
+    private String createJsonPayloadFromOauthApplication(OAuthApplicationInfo oAuthApplicationInfo,
+                                                         Map<String, Object> paramMap) throws APIManagementException {
+        String clientName = oAuthApplicationInfo.getClientName();
+        if (log.isDebugEnabled()) {
+            log.debug(String.format("Creating json payload from Oauth application info for the application: %s",
+                    clientName));
+        }
+        if (StringUtils.isNotEmpty(clientName)) {
+            paramMap.put(OktaConstants.CLIENT_NAME, clientName);
+        }
+
+        String clientRedirectUri = oAuthApplicationInfo.getCallBackURL();
+        if (StringUtils.isEmpty(clientRedirectUri)) {
+            handleException("Mandatory parameter CallBack URL is missing");
+        }
+        List<String> redirectUris = Collections.singletonList(clientRedirectUri);
+        paramMap.put(OktaConstants.CLIENT_REDIRECT_URIS, redirectUris);
+
+        Object clientResponseTypes = oAuthApplicationInfo.getParameter(OktaConstants.CLIENT_RESPONSE_TYPES);
+        if (clientResponseTypes != null) {
+            String[] responseTypes = ((String) clientResponseTypes).split(",");
+            JSONArray jsonArray = new JSONArray();
+            Collections.addAll(jsonArray, responseTypes);
+            paramMap.put(OktaConstants.CLIENT_RESPONSE_TYPES, jsonArray);
+        } else {
+            handleException("Mandatory parameter response_types is missing");
+        }
+
+        Object clientGrantTypes = oAuthApplicationInfo.getParameter(OktaConstants.CLIENT_GRANT_TYPES);
+        if (clientGrantTypes != null) {
+            String[] grantTypes = ((String) clientGrantTypes).split(",");
+            JSONArray jsonArray = new JSONArray();
+            Collections.addAll(jsonArray, grantTypes);
+            paramMap.put(OktaConstants.CLIENT_GRANT_TYPES, jsonArray);
+        } else {
+            handleException("Mandatory parameter grant_types is missing");
+        }
+
+        Object clientPostLogoutRedirectUris = oAuthApplicationInfo.getParameter(
+                OktaConstants.CLIENT_POST_LOGOUT_REDIRECT_URIS);
+        if (clientPostLogoutRedirectUris != null) {
+            String[] postLogoutRedirectUris = ((String) clientPostLogoutRedirectUris).split(",");
+            JSONArray jsonArray = new JSONArray();
+            Collections.addAll(jsonArray, postLogoutRedirectUris);
+            paramMap.put(OktaConstants.CLIENT_POST_LOGOUT_REDIRECT_URIS, jsonArray);
+        }
+
+        String tokenEndpointAuthMethod = (String) oAuthApplicationInfo.getParameter(
+                OktaConstants.CLIENT_TOKEN_ENDPOINT_AUTH_METHOD);
+        if (StringUtils.isNotEmpty(tokenEndpointAuthMethod)) {
```

Comment:
Shall we do something like this for the validation?

void validate(Parameter) {
  if(StringUtils.isEmpty(parameter) {
        throw exception();
}

Then in this method we will only have validate(parameter) without if condition. If it fails next line won't be executing. Can we do this for all the mandatory parameters

Reply:
If we do like this then we cant specify which parameter (parameter name) is missing in the error message.